src/Regridder.jl

```diff
@@ -638,33 +638,59 @@ function cgll2latlonz(field; DIR = "cgll2latlonz_dir", nlat = 360, nlon = 720, c
 end
 
 """
-    truncate_dataset(datafile, name, datapath_trunc, date0, time_start, time_end, comms_ctx)
+    truncate_dataset(datafile, varname, datapath_trunc, date0, t_start, t_end, comms_ctx)
 
 Truncates given data set, and constructs a new dataset containing only the dates that are used in the simulation
 """
 function truncate_dataset(
     datafile,
-    name,
+    varname,
     datapath_trunc,
     date0,
-    time_start,
-    time_end,
+    t_start,
+    t_end,
     comms_ctx::ClimaComms.AbstractCommsContext,
 )
-    date_start = date0 + Dates.Second(time_start)
-    date_end = date0 + Dates.Second(time_start + time_end)
+    date_start = date0 + Dates.Second(t_start)
+    date_end = date0 + Dates.Second(t_start + t_end)
 
-    file_name = replace(string(name, "_truncated_data_", string(date_start), string(date_end), ".nc"), r":" => "")
+    file_name =
+        replace(string(lowercase(varname), "_truncated_data_", string(date_start), string(date_end), ".nc"), r":" => "")
     datafile_truncated = joinpath(datapath_trunc, file_name)
 
     if ClimaComms.iamroot(comms_ctx)
         ds = NCDatasets.NCDataset(datafile, "r")
         dates = ds["time"][:]
 
+        # Find the bounding indices of the dates we need
         (start_id, end_id) = find_idx_bounding_dates(dates, date_start, date_end)
 
+        var_truncated = NCDatasets.nomissing(ds[varname][:, :, start_id:end_id])
```

Comment:
This is assuming a certain number of dimensions. Could this be generalized so it always indexes into time, e.g. even for 4D arrays?

Reply:
Good point! I tried using `view` and selecting for the time instead, and it works locally